### DISCUSSION TITLE
Merging graphs

### DISCUSSION THREAD

**Discussion by alexandernst** [clientIO/joint#2208] [Q&A] [answered]
### Introduction

I have a graph (lets call it `Foo`) with two nodes connected by a link, like this:

```
A: id 1 -----> L: id 2 -----> B: id 3
```

Lets say I have another graph (lets call it `Bar`) that has some cells connected by links. These cells might or might not have the IDs.

For the purpose of this question lets asume that the cells in `Bar` are the same as the cells in `Foo` and that they have the same IDs, which means that they will collide with the cells in `Foo`. How can I get all the cells from `Bar` and insert them in `Foo`, changing whichever IDs are colliding, while keeping all the connections / links?

The end result would look like:

```
A: id 1 -----> L: id 2 -----> B: id 3
A: id 2345683746 -----> L: id 546845689 -----> B: id 234578999433
```

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
You can clone the other graph with [dia.Graph.prototype.cloneSubgraph()](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.Graph.prototype.cloneSubgraph).
```js
graph1.addCells(Object.values(graph2.cloneSubgraph(graph2.getCells())));
```

**Reply by alexandernst** [reply]
I finished implementing and testing this. It works like a charm! 👏 thank you!
